hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -281,7 +313,18 @@ private boolean writeUpdateRecord(HoodieRecord<T> hoodieRecord, GenericRecord ol
         return false;
       }
     }
-    return writeRecord(hoodieRecord, indexedRecord, isDelete);
+    boolean result = writeRecord(hoodieRecord, indexedRecord, isDelete);
+    if (cdcEnabled) {
+      if (indexedRecord.isPresent()) {
+        GenericRecord record = (GenericRecord) indexedRecord.get();
+        cdcData.add(cdcRecord(CDCOperationEnum.UPDATE, hoodieRecord.getRecordKey(), hoodieRecord.getPartitionPath(),
```

Comment:
IMO, it's ok. 
A base parquet file is about 128M in most common cases. Even if all the records is updated, the `cdcData` will take the memory that is less that about 300M. And if the workflow is heavy, user can increase the memory of workers.
But If we are worry about this, use the `hudi.common.util.collection.ExternalSpillableMap` instead of this.

Reply:
In general, with all the java/jvm overhead, I think. it'll be more than 300M comfortably. Can we use the spillable map instead here in this PR

Reply:
ok, i will use `hudi.common.util.collection.ExternalSpillableMap` instead here.